main.go

```diff
@@ -194,13 +194,21 @@ func main() {
 
 	mux.Handle(apiPath+"/v1/", http.StripPrefix(apiPath+"/v1", av1))
 
-	go closeListenerOnQuit(l, quitCh, logger)
-	err = web.Serve(l, &http.Server{Addr: *listenAddress, Handler: mux}, *webConfig, logger)
-	level.Error(logger).Log("msg", "HTTP server stopped", "err", err)
-	// To give running connections a chance to submit their payload, we wait
-	// for 1sec, but we don't want to wait long (e.g. until all connections
-	// are done) to not delay the shutdown.
-	time.Sleep(time.Second)
+	server := &http.Server{
+		Addr:    *listenAddress,
+		Handler: mux,
+	}
+
+	go shutdownServerOnQuit(server, quitCh, logger)
+	err = web.Serve(l, server, *webConfig, logger)
+
+	// In the case of a shutdown, log gracefully
```

Comment:
Nit: End sentence with a period.

And, even more nitpicking, "log gracefully" might not really nail it. How about "In the case of a graceful shutdown, do not log the error."